### PR TITLE
Docs updates

### PR DESCRIPTION
**Pull request by alex-zenml**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
Docs updates as discussed.

### CODE REVIEW

File: docs/book/core-concepts.md [hunk 1]
```diff
@@ -152,6 +154,10 @@ zenml stack register STACK_NAME \
     -o ORCHESTRATOR_NAME
 ```
 
+**Container Registry**
+
+Container registries are where pre-built (Docker) containers are hosted, ready for use in various configurations. A ZenML workflow involving a container registry would see you spinning up a Kubernetes cluster and then deploying a pipeline to be run on Kubeflow Pipelines. As part of the deployment to the cluster, the ZenML base image would be downloaded and used as the basis for the deployed 'run'. When you are running a local Kubeflow stack, you will also have a local container registry which stores the container images you create that bundle up your pipeline code.
```
Comment (htahir1) [inline docs/book/core-concepts.md:159]: Probably makes sense to make this a bit more general. A container registry is basically a store for docker images. The local variant of this is used to push a *new* docker image that is created based off a base image  or custom image of your choice. However in the future there can be a cloud variant which points to hosted registries for example the Google container registry or dockerhub

File: docs/book/core-concepts.md [hunk 2]
```diff
@@ -104,7 +104,9 @@ Artifacts can be serialized and deserialized (i.e. written and read from the Art
 
 **Materializers**
 
-A materializer defines how and where Artifacts live in between steps.
+A materializer defines how and where Artifacts live in between steps. It is used to convert a ZenML artifact into a specific format. They are most often used to handle the input or output of ZenML steps, and can be extended by building on the `BaseMaterializer` class. We care about this because steps are not just isolated pieces of work; they are linked together and the outputs of one step might well be the inputs of the next.
+
+We have some built-in ways to serialize and deserialize the data flowing between steps. Of course, if you are using some library or tool which doesn't work with our built-in options, you can write [your own custom materializer](https://docs.zenml.io/guides/low-level-api/chapter-5) to ensure that your data can be passed from step to step in this way.
```
Comment (htahir1) [inline docs/book/core-concepts.md:109]: Maybe write here about the fileio role in this. I.e using the fileio we ensure it works across artifact stores?

**Review by htahir1 — COMMENTED** [review]
Some suggestions but going in the right direction!

File: docs/book/core-concepts.md (hunk 1 above)
Comment (schustmi) [reply docs/book/core-concepts.md:159]: The *newly* created images are pushed to the container registry specified in the stack, this however is not limited to a local registry (even now, it is entirely possible to specify a gcr.io/... registry URL as long as the local docker client is authorized to push to that registry).

**Comment by alex-zenml** [comment]
@htahir1 should be better now

**Review by htahir1 — APPROVED** [review]

**Pull request merged**